modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/components/sidebar/sections/assignments/select-assignment/Role.js

```diff
@@ -89,79 +35,17 @@ const Roles = () => {
 		}));
 	};
 
-	const filterItems = () =>
-		resource.items.filter((item) =>
-			!filter
-				? item
-				: item.name.toLowerCase().match(fieldValues.name.toLowerCase())
-		);
-
 	return (
-		<SidebarPanel panelTitle={Liferay.Language.get('select-role')}>
-			<ClayForm.Group>
-				<ClayAutocomplete>
-					<label htmlFor="role-name">
-						{Liferay.Language.get('role')}
-
-						<span className="ml-1 mr-1 text-warning">*</span>
-					</label>
-
-					<ClayAutocomplete.Input
-						autoComplete="off"
-						id="role-name"
-						onChange={(event) => handleInputChange(event)}
-						onFocus={() => handleInputFocus()}
-						placeholder="Search"
-						value={fieldValues.name}
-					/>
-
-					<ClayAutocomplete.DropDown
-						active={(!!resource && active) || initialLoading}
-						closeOnClickOutside
-						onSetActive={setActive}
-					>
-						<ClayDropDown.ItemList>
-							{(error || (resource && resource.error)) && (
-								<ClayDropDown.Item className="disabled">
-									{Liferay.Language.get('no-results-found')}
-								</ClayDropDown.Item>
-							)}
-
-							{!error &&
-								resource?.items &&
-								filterItems().map((item) => (
-									<ClayAutocomplete.Item
-										key={item.id}
-										onClickCapture={() => {
-											handleItemClick(item);
-										}}
-										value={item.name}
-									/>
-								))}
-						</ClayDropDown.ItemList>
-					</ClayAutocomplete.DropDown>
-
-					{loading && <ClayAutocomplete.LoadingIndicator />}
-				</ClayAutocomplete>
-			</ClayForm.Group>
-
-			<ClayForm.Group>
-				<label htmlFor="role-id">
-					{Liferay.Language.get('role-id')}
-
-					<span className="ml-1 mr-1 text-warning">*</span>
-				</label>
-
-				<ClayInput
-					component="input"
-					disabled
-					id="role-id"
-					type="text"
-					value={fieldValues.id}
-				/>
-			</ClayForm.Group>
-		</SidebarPanel>
+		<BaseRole
+			defaultFieldValue={{
+				id: selectedItem.data.assignments?.sectionsData?.id || '',
+				name: selectedItem.data.assignments?.sectionsData?.name || '',
+			}}
+			inputLabel={Liferay.Language.get('role-id')}
+			selectLabel={Liferay.Language.get('role')}
+			updateSelectedItem={updateSelectedItem}
+		></BaseRole>
```

Comment:
@eliasantosf You don't need to open and close the BaseRole tag. You can use <BaseRole />